pkgs/development/interpreters/python/cpython/3.7/default.nix

```diff
@@ -63,6 +63,12 @@ in stdenv.mkDerivation {
 
   patches = [
     ./no-ldconfig.patch
+
+    # Fixes darwin build.
+    (fetchpatch {
+      url = https://github.com/LnL7/cpython/commit/6750bb7abfb13775e1bae3143099ead4dea68d6a.patch;
```

Comment:
Why do you download the patch from your fork instead of including it in Nixpkgs? This makes it harder for others to change it.

It seems that this patch is better (and may be proposed for inclusion upstream):
```
 #ifdef HAVE_LIBUTIL_H
 #include <libutil.h>
-#else
+#endif /* HAVE_LIBUTIL_H */
 #ifdef HAVE_UTIL_H
 #include <util.h>
 #endif /* HAVE_UTIL_H */
-#endif /* HAVE_LIBUTIL_H */
```

Reply:
There's an issue and pull request for it, if that's merged we can use upstream.

Reply:
Thanks, I have found the issue: https://bugs.python.org/issue34027

Reply:
And the PR: https://github.com/python/cpython/pull/8056

Reply:
Can you include the link to the pull request, so it is easier to track, when it is merged?

Reply:
I opened this in preparation of https://github.com/NixOS/nixpkgs/pull/42777, it's not urgent until that's merged. I'll update it with the relevant information when upstream makes a decision.

Reply:
Updated to use @orivej's patch, since that's most likely to get accepted upstream.